src/raft.c

```diff
@@ -341,24 +361,25 @@ void RaftExecuteCommandArray(RedisRaftCtx *rr,
         int old_entered_eval = rr->entered_eval;
 
         if ((cmdlen == 4 && !strncasecmp(cmd, "eval", 4)) ||
-            (cmdlen == 7 && !strncasecmp(cmd, "evalsha", 7))) {
+            (cmdlen == 7 && !strncasecmp(cmd, "evalsha", 7)) ||
+            (cmdlen == 5 && !strncasecmp(cmd, "fcall", 5))) {
             rr->entered_eval = 1;
         }
 
+        /* Explanation:
+         * When we have an ACL, we will have a user set on the context, so need "C"
+         */
+        char *resp_call_fmt = cmds->acl ? "CE0v" : "E0v";
+
         enterRedisModuleCall();
-        reply = RedisModule_Call(ctx, cmd,
-                                 rr->resp_call_fmt, &c->argv[1], c->argc - 1);
-        int ret_errno = errno;
+        RedisModule_SetContextUser(ctx, user);
+        reply = RedisModule_Call(ctx, cmd, resp_call_fmt, &c->argv[1], c->argc - 1);
         exitRedisModuleCall();
 
         rr->entered_eval = old_entered_eval;
 
         if (reply_ctx) {
-            if (reply) {
-                RedisModule_ReplyWithCallReply(reply_ctx, reply);
-            } else {
-                replyRMCallError(reply_ctx, ret_errno, cmd, cmdlen);
-            }
+            RedisModule_ReplyWithCallReply(ctx, reply);
```

Comment:
```suggestion
            RedisModule_ReplyWithCallReply(reply_ctx, reply);
```
Although in the current implementation having `reply_ctx` means `ctx == reply_ctx` I think we should use  `reply_ctx` for readability and future uses.